src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -122,15 +124,20 @@ public CloudWatchAttempt processLogFiles(ComponentLogFileInformation componentLo
                 .replace("{componentName}", componentLogFileInformation.getName());
         attempt.setLogGroupName(logGroupName);
         String logStreamName = getLogStreamName(thingName);
-
+        LogFileGroup logFileGroup = componentLogFileInformation.getLogFileGroup();
         // Run the loop until all the log files from the component have been read or the max message
         // size has been reached.
         while (!componentLogFileInformation.getLogFileInformationList().isEmpty() && !reachedMaxSize.get()) {
-            LogFile logFile = componentLogFileInformation.getLogFileInformationList().get(0).getLogFile();
-            long startPosition = componentLogFileInformation.getLogFileInformationList().get(0).getStartPosition();
-            //TODO Optional<String> fileHash = componentLogFileInformation.getLogFileInformationList().get(0)
-            // .getFileHash();
+            LogFileInformation tempLogFileInfomation = componentLogFileInformation.getLogFileInformationList().get(0);
+            LogFile logFile = tempLogFileInfomation.getLogFile();
+            long startPosition = tempLogFileInfomation.getStartPosition();
+            String fileHash = tempLogFileInfomation.getFileHash();
             String fileName = logFile.getAbsolutePath();
+            if ("".equals(fileHash)) {
```

Comment:
Avoid exposing this. Do instead

```java
if (logFile.hasContents()) { // that logic would be inside the log file, then you are giving more meaning to it and also we can change our hashing strategy later on.
}
```

Reply:
This scenario shouldn't even happen. Add a TODO here to ensure that we only ever add files that have content to the `componentLogFileInformation`  the we don't have to be mutating the list here